reconcile/saas_auto_promotions_manager/subscriber.py

```diff
@@ -45,6 +45,16 @@ def __init__(
         self.namespace_file_path = namespace_file_path
         self._content_hash = ""
 
+    def has_diff(self) -> bool:
+        current_hashes: list[ConfigHash] = []
```

Comment:
since `current_hashes` only used as `set`, can define it as `set` directly. Can use set comprehension or flatten by itertools

```python
current_hashes = {el for s in self.config_hashes_by_channel_name.values() for el in s}
```

or 

```python
current_hashes = set(itertools.chain.from_iterable(self.config_hashes_by_channel_name.values()))
```

Reply:
Done :+1: